compileopts/target.go

```diff
@@ -40,6 +40,7 @@ type TargetSpec struct {
 	LDFlags          []string `json:"ldflags"`
 	LinkerScript     string   `json:"linkerscript"`
 	ExtraFiles       []string `json:"extra-files"`
+	ElfPatches       []string `json:"elf-patches"`              // Patches that should be applied to the ELF after linking
```

Comment:
I'm not entirely convinced this level of abstraction is the right level. While we might add more patches in the future, I think a simple boolean (like `automatic-stack-size`, for example `rp2040-boot-patch`) would be simpler to understand and modify. It's always possible to add a slice of patches in the future.

Alternatively, you could leave out a change to the target JSON out entirely and instead use a specially named section (like `.rp2040_boot2`) which will be modified if present.

Reply:
I'm less certain about special section names - means the build logic has to open the elf for all targets to see if the section exists?  For now I've gone with the suggestion of `rp2040-boot-patch` in target spec.